integration-tests/helpers/fake-agent.js

```diff
@@ -0,0 +1,266 @@
+'use strict'
+
+const { createHash } = require('crypto')
+const EventEmitter = require('events')
+const http = require('http')
+const uuid = require('crypto-randomuuid')
+const express = require('express')
+const bodyParser = require('body-parser')
+const msgpack = require('msgpack-lite')
+const codec = msgpack.createCodec({ int64: true })
+const upload = require('multer')()
+
+module.exports = class FakeAgent extends EventEmitter {
+  constructor (port = 0) {
+    super()
+    this.port = port
+    this._rc_files = []
+  }
+
+  async start () {
+    return new Promise((resolve, reject) => {
+      const timeoutObj = setTimeout(() => {
+        reject(new Error('agent timed out starting up'))
+      }, 10000)
+      this.server = http.createServer(buildExpressServer(this))
+      this.server.on('error', reject)
+      this.server.listen(this.port, () => {
+        this.port = this.server.address().port
+        clearTimeout(timeoutObj)
+        resolve(this)
+      })
+    })
+  }
+
+  stop () {
+    return new Promise((resolve) => {
+      this.server.on('close', resolve)
+      this.server.close()
+    })
+  }
+
+  /**
+   * Remove any existing config added by calls to FakeAgent#addRemoteConfig.
+   */
+  resetRemoteConfig () {
+    this._rc_files = []
+  }
+
+  /**
+   * Add a config object to be returned by the fake Remote Config endpoint.
+   * @param {Object} config - Object containing the Remote Config "file" and metadata
+   * @param {number} [config.orgId=2] - The Datadog organization ID
+   * @param {string} config.product - The Remote Config product name
+   * @param {string} config.id - The Remote Config config ID
+   * @param {string} [config.name] - The Remote Config "name". Defaults to the sha256 hash of `config.id`
+   * @param {Object} config.config - The Remote Config "file" object
+   */
+  addRemoteConfig (config) {
+    config = { ...config }
```

Comment:
nit: isn't Object.assign() faster ? :p

Reply:
I've seen benchmarks suggesting the opposite, so I don't think so. But in any case:

1. If there's a performance difference, it's only noticeable for very large objects, so not worth optimizing for in this case.
2. This is a test and not a hot code-path, so we should always opt for the more modern syntax if possible.

But in general, unless there's a huge difference (which is usually because of a bug in the interpreter), it's always better to not try and optimize your code against a specific version of V8. V8 changes over time and these things will improve. If you optimize for a quirk in V8, you end up loosing as V8 will later fix the performance degradation and your code is now slower than it could have been if you didn't try to optimize it. Just my 5 cents 😅